### PR TITLE
Update typings for AngularJS 1.8.1 (packages: `angular`, `angular-sanitize`)

### PR DESCRIPTION
**Pull request by mgol**
AngularJS 1.8.1 added alternative APIs for ones that may be considered offensive
and deprecated the old ones:
https://github.com/angular/angular.js/blob/master/CHANGELOG.md#181-mutually-supporting-2020-09-30

The `IHttpProvider` type was outdated as well. The `useLegacyPromiseExtensions`
method has been removed in AngularJS 1.6.0:
https://code.angularjs.org/1.5.11/docs/api/ng/provider/$httpProvider#useLegacyPromiseExtensions
`xsrfWhitelistedOrigins` and `xsrfTrustedOrigins` were missing:
https://docs.angularjs.org/api/ng/provider/$httpProvider#xsrfTrustedOrigins

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/angular/angular.js/blob/master/CHANGELOG.md#181-mutually-supporting-2020-09-30, https://docs.angularjs.org/api/ng/provider/$httpProvider#xsrfTrustedOrigins
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## angular-sanitize ([<kbd>unpkg</kbd>](https://unpkg.com/browse/angular-sanitize@latest/))
was missing the following properties:
1. padStart
1. padEnd
1. trimLeft
1. trimRight
1. trimStart

<details>
<summary>as well as these 1 other properties...</summary>
<p>trimEnd</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 753cdb3bd4c27ade5335dcc067a0aadd76a05283
</p>

**Review by peterblazejewicz — APPROVED** [review]
@mgol thx for the update!
LGTM!

**Comment by mgol** [comment]
@peterblazejewicz what's the procedure now? Do I just wait for a DefinitelyTyped maintainer to notice this PR and do a review? Should I split the `angular-sanitize` changes out to a separate PR?

**Comment by peterblazejewicz** [comment]
@mgol 
So, because you've not included tests for your changes, there is no quick path here (the other pair of eyes from the community need to review the PR, the tests are added/amended, the CIs pass, etc).
So either wait for DT maintainer to arrive and review that one (it's already labelled with 'owner approved', so it would be quick), or amend your PR to incorporate tests for both packages. Thx!

**Review by falsyvalues — APPROVED** [review]
@mgol Looks good. Would be good to add tests that covers changes.

File: types/angular-sanitize/index.d.ts
Comment (thorn0) [inline types/angular-sanitize/index.d.ts:26]:
```suggestion
         * URLs allowed in attribute values are configured using the methods aHrefSanitizationTrustedUri and imgSrcSanitizationTrustedUri of $compileProvider.
```

File: types/angular/index.d.ts
Comment (thorn0) [inline types/angular/index.d.ts:1413]:
```suggestion
        /** @deprecated The old name of aHrefSanitizationTrustedUri. Kept for compatibility. */
```

File: types/angular/index.d.ts
Comment (thorn0) [inline types/angular/index.d.ts:1415]:
```suggestion
        /** @deprecated The old name of aHrefSanitizationTrustedUri. Kept for compatibility. */
```

File: types/angular/index.d.ts
Comment (thorn0) [inline types/angular/index.d.ts:1421]:
```suggestion
        /** @deprecated The old name of imgSrcSanitizationTrustedUri. Kept for compatibility. */
```

File: types/angular/index.d.ts
Comment (thorn0) [inline types/angular/index.d.ts:1423]:
```suggestion
        /** @deprecated The old name of imgSrcSanitizationTrustedUri. Kept for compatibility. */
```

File: types/angular/index.d.ts
Comment (thorn0) [inline types/angular/index.d.ts:1813]:
```suggestion
        /** @deprecated The old name of xsrfTrustedOrigins. Kept for compatibility. */
```

**Review by thorn0 — CHANGES_REQUESTED** [review]

**Review by thorn0 — APPROVED** [review]

**Comment by mgol** [comment]
Comments addressed, apart from tests. It turns out AngularJS changelog is incorrect & some of the APIs have different name than the changelog specifies: https://github.com/angular/angular.js/issues/17088. I changed them.

I'll look into adding some tests later.

**Comment by orta** [comment]
Classing as acceptable from the DT maintainers perspective, which means it can be self-merged in the future after the tests are added 👍🏻

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by mgol** [comment]
I rebased & added a new commit with tests for the `angular` changes. I'd appreciate if someone looked at those tests before this gets merged.

**Review by peterblazejewicz — APPROVED** [review]

**Comment by mgol** [comment]
Ready to merge

**Review by andrewbranch — APPROVED** [review]

**Pull request merged**
